HW04.Rmd

```diff
@@ -1,126 +1,306 @@
+
+#initial look at data
+glimpse(gun_deaths)
+
+#change month col (numbers) to month names
+gun_deaths$month <- factor(gun_deaths$month, labels = c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"))
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+
+deaths_per_month <- gun_deaths %>%
+    group_by(month) %>%
+    count()
+
+kable(deaths_per_month, format = "markdown",
+      col.names = c("Month", "Number of Deaths"),
+      caption = "Number of gun related deaths per month.")
+
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+#create plot
+ggplot(gun_deaths, aes(month)) +
+    geom_bar(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 9000, by = 1000)) +
+    
+#add labels
+    labs(title = "Gun Deaths per Month from 2012-2014",
+         x     = "Month",
+         y     = "Number of Deaths") +
+    
+ #add theme   
+    theme_classic()
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+#create plot
+    #fct_infreq(): orders by number of observations with each level (largest first)
+ggplot(gun_deaths, aes(fct_infreq(intent))) +
+    geom_bar(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 60000, by = 10000)) +
+    
+#add labels
+    labs(title = "Intent of Gun Related Deaths from 2012-2014",
+         x     = "Intent",
+         y     = "Number of Deaths") +
+    
+ #add theme   
+    theme_classic()
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+#create new data frame to remove NA age
+gun_deaths_by_sex <- gun_deaths %>%
+    filter(age != "Not Available") %>%
+    mutate(age = as.numeric(age))
+
+# generating a boxplot visualizing the data
+ggplot(gun_deaths_by_sex, aes(sex, age)) +
+    geom_boxplot(fill = "cornflowerblue") +
+    scale_y_continuous(breaks = seq(0, 100, by = 10)) +
+
+#add labels
+    labs(title = "Gun Related Deaths by Sex from 2012-2014",
+         x     = "Sex",
+         y     = "Age (years)") +
+    
+ #add theme   
+    theme_classic()
+
+
+#find average age of female gun death victims and create a new data frame
+female_avg_age <- gun_deaths %>%
+    group_by(sex) %>%
+    filter(sex == 'F') %>%
+    mutate(age = as.numeric(age)) %>%
+    summarize(average_age_female = mean(age, na.rm = TRUE)) # na.rm removes NA values
```

Comment:
I've never heard of the na.rm argument but I believe it will definitely come handy in the future. Thanks for introducing this! :)